include/partitioner/multi_level_graph.hpp

```diff
@@ -146,6 +149,14 @@ class MultiLevelGraph : public util::StaticGraph<EdgeDataT, Ownership>
         return max_border_node_id;
     }
 
+    auto data() &&
```

Comment:
This is a kind of weird mechanism. Do you think we could also solve this by protected access to these members by the sub-class? Or is this a problem with templates?

Reply:
It is not directly a problem with templates but with different `EdgeBasedGraphEdgeData` template arguments of customizer and partitioner graphs and `customizer::MultiLevelGraph` is not allowed to access protected members  of `partitioner::MultiLevelGraph`.

An rvalue qualifier `&&` is just a safety belt to prevent using an object instance after `graph.data()` call and explicit requirement for `std::move(graph).data()`. `&&` here and `std::move` at the calling place can be removed.